packages/ui/src/components/va-pagination/VaPagination.vue

```diff
@@ -1,53 +1,52 @@
 <template>
-  <va-button-group
+  <nav
     v-if="showPagination"
     class="va-pagination"
-    :flat="$props.flat"
-    :size="$props.size"
-    :color="$props.color"
-    outline
-    @keydown.left.stop="onUserInput(currentValue - 1)"
-    @keydown.right.stop="onUserInput(currentValue + 1)"
+    aria-label="pagination"
+    :class="classComputed"
+    @keydown.left.stop="goPrevPage"
+    @keydown.right.stop="goNextPage"
+    @keydown.up.stop="goPrevPage"
+    @keydown.down.stop="goNextPage"
   >
     <va-button
       v-if="showBoundaryLinks"
-      aria-label="go first page"
+      aria-label="go to the first page"
       :disabled="$props.disabled || currentValue === 1"
       :icon="$props.boundaryIconLeft"
+      v-bind="buttonPropsComputed"
       @click="onUserInput(1)"
     />
     <va-button
       v-if="showDirectionLinks"
-      aria-label="go prev page"
-      outline
+      aria-label="go to the previous page"
       :disabled="$props.disabled || currentValue === 1"
       :icon="$props.directionIconLeft"
-      @click="onUserInput(currentValue - 1)"
+      v-bind="buttonPropsComputed"
+      @click="goPrevPage"
     />
     <slot v-if="!$props.input">
       <va-button
         v-for="(n, i) in paginationRange"
-        :key="i"
-        class="va-pagination__numeric-button"
-        outline
-        :aria-label="`go to ${n} page`"
-        :aria-current="n === currentValue"
-        :style="activeButtonStyle(n)"
-        :disabled="$props.disabled || n === '...'"
+        :key="`pagination-page-${i}`"
         :class="{ 'va-button--ellipsis': n === '...'}"
+        :aria-label="`go to the ${n} page`"
+        :aria-current="n === currentValue"
+        :disabled="$props.disabled || ['...', currentValue].includes(n)"
+        v-bind="getPageButtonProps(n)"
         @click="onUserInput(n)"
       >
         {{ n }}
       </va-button>
     </slot>
     <input
       v-else
+      v-model="inputValue"
       ref="htmlInput"
       class="va-pagination__input va-button"
       aria-label="enter the page number to go"
```

Comment:
Hmm, this is necessary and we don't allow to pass translation?

Reply:
Will move `aria-attributes translations` to different issue - it's not related only to this component.